common/viperutil/config_util.go

```diff
@@ -303,7 +304,7 @@ func bccspHook(f reflect.Type, t reflect.Type, data interface{}) (interface{}, e
 
 	config := factory.GetDefaultOpts()
 
-	err := mapstructure.Decode(data, config)
+	err := mapstructure.WeakDecode(data, config)
```

Comment:
The one bit you may want to consider is whether or not to carry the `ErrorUnused` option here.

Reply:
🤔I think of it as extra validation but not sure if it harms or helps.